[gperf] build failure

### Operating system

Linux

### Compiler

Clang

### Steps to reproduce the behavior

```Shell
LDFLAGS="-fuse-ld=lld" CC=clang-16 CXX=clang++-16 CXXFLAGS="-stdlib=libc++ -fPIC" ./vcpkg install gperf:x64-linux
```


### Failure logs

` /home/hv/libs/vcpkg/buildtrees/gperf/src/gperf-3-d4135ad4b1.clean/lib/getline.cc:58:7: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
      register int c = getc (stream);
      ^~~~~~~~~
1 error generated.
[3/18] /usr/bin/clang-16  -I/home/hv/libs/vcpkg/buildtrees/gperf/src/gperf-3-d4135ad4b1.clean/lib -fPIC -g -MD -MT CMakeFiles/gp.dir/lib/getopt1.c.o -MF CMakeFiles/gp.dir/lib/getopt1.c.o.d -o CMakeFiles/gp.dir/lib/getopt1.c.o -c /home/hv/libs/vcpkg/buildtrees/gperf/src/gperf-3-d4135ad4b1.clean/lib/getopt1.c `

### Additional context

_No response_

--- a/ports/gperf/portfile.cmake
+++ b/ports/gperf/portfile.cmake
@@ -9,6 +9,8 @@ vcpkg_download_distfile(ARCHIVE
 vcpkg_extract_source_archive(
     SOURCE_PATH
     ARCHIVE ${ARCHIVE}
+    PATCHES 
+    	remove_register_keyword_cpp17.patch
 )
 
 file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})

--- a/ports/gperf/remove_register_keyword_cpp17.patch
+++ b/ports/gperf/remove_register_keyword_cpp17.patch
@@ -1,0 +1,19 @@
+commit a5ff738a4b545ad25640753b651a6322b463c7b0
+Author: rdh <ridwanabdulhafidh@gmail.com>
+Date:   Wed Aug 2 05:04:58 2023 -0400
+
+    update cpp
+
+diff --git a/lib/getline.cc b/lib/getline.cc
+index c57c633..0984a7c 100644
+--- a/lib/getline.cc
++++ b/lib/getline.cc
+@@ -55,7 +55,7 @@ getstr (char **lineptr, size_t *n, FILE *stream, char terminator, size_t offset)
+ 
+   for (;;)
+     {
+-      register int c = getc (stream);
++      int c = getc (stream);
+ 
+       /* We always want at least one char left in the buffer, since we
+          always (unless we get an error while reading the first char)

--- a/ports/gperf/remove_register_keyword_cpp17.patch
+++ b/ports/gperf/remove_register_keyword_cpp17.patch
@@ -1,9 +1,3 @@
-commit a5ff738a4b545ad25640753b651a6322b463c7b0
-Author: rdh <ridwanabdulhafidh@gmail.com>
-Date:   Wed Aug 2 05:04:58 2023 -0400
-
-    update cpp
-
 diff --git a/lib/getline.cc b/lib/getline.cc
 index c57c633..0984a7c 100644
 --- a/lib/getline.cc

--- a/ports/gperf/vcpkg.json
+++ b/ports/gperf/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "gperf",
   "version": "3.1",
-  "port-version": 5,
+  "port-version": 6,
   "description": "GNU perfect hash function generator",
   "homepage": "https://www.gnu.org/software/gperf/",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2966,7 +2966,7 @@
     },
     "gperf": {
       "baseline": "3.1",
-      "port-version": 5
+      "port-version": 6
     },
     "gperftools": {
       "baseline": "2019-09-02",

--- a/versions/g-/gperf.json
+++ b/versions/g-/gperf.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "ef4a25070d082723213d25e535499902db1e3731",
+      "git-tree": "f2b89ce69481e9a4b5c9c26b49e92c38c2de9561",
       "version": "3.1",
       "port-version": 6
     },

--- a/versions/g-/gperf.json
+++ b/versions/g-/gperf.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "f2b89ce69481e9a4b5c9c26b49e92c38c2de9561",
+      "git-tree": "590e19515bff5b0abc6d9f73ba20a19a71555a32",
       "version": "3.1",
       "port-version": 6
     },

--- a/versions/g-/gperf.json
+++ b/versions/g-/gperf.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "ef4a25070d082723213d25e535499902db1e3731",
+      "version": "3.1",
+      "port-version": 6
+    },
+    {
       "git-tree": "c1dc7145f2a0607798bfdaede7bf064b88e908ac",
       "version": "3.1",
       "port-version": 5

